src/Job/ToolRunner.ts

```diff
@@ -52,27 +53,49 @@ export class ToolRunner {
   tag = this.constructor.name;  // logging tag
 
   // This variable is undefined while a prcess is not running
-  private child: cp.ChildProcessWithoutNullStreams|undefined = undefined;
+  private child: cp.ChildProcess|undefined = undefined;
 
   // When the spawned process was killed by kill() method, set this true
   // This value must be set to false when starting a process
   private killedByMe = false;
 
   private handlePromise(
       resolve: (value: SuccessResult|PromiseLike<SuccessResult>) => void,
-      reject: (value: ErrorResult|PromiseLike<ErrorResult>) => void) {
+      reject: (value: ErrorResult|PromiseLike<ErrorResult>) => void, root: boolean) {
     // stdout
-    this.child!.stdout.on(K_DATA, (data: any) => {
+    this.child!.stdout!.on(K_DATA, (data: any) => {
       Logger.append(data.toString());
     });
     // stderr
-    this.child!.stderr.on(K_DATA, (data: any) => {
+    this.child!.stderr!.on(K_DATA, (data: any) => {
       Logger.append(data.toString());
     });
 
     this.child!.on(K_EXIT, (code: number|null, signal: NodeJS.Signals|null) => {
       this.child = undefined;
 
+      if (root) {
+        /*
+        NOTE
+
+        Considering a case:
+
+          `echo correct_pw | sudo -S ...` --> `echo wrong_pw | sudo -S ...`
+
+        The second call does not fail because pw is cached by sudo.
+        Let's invalidate sudo cache by putting `sudo -k` somewhere.
+
+          `echo correct_pw | sudo -S ...` --> `sudo -k` --> `echo wrong_pw | sudo -S ...`
+
+        When should we call `sudo -k`?
+        1. We observed that `sudo -k` right after spawning `echo correct_pw | sudo -S ...` does not
+          seem to work, making `echo wrong_pw | sudo -S ...` succeed.
+        2. Calling `sudo -k` after first sudo process finishes seems to make
+          `echo wrong_pw | sudo -S ...` fail.
+        */
+        cp.spawnSync('sudo', ['-k']);
```

Comment:
```suggestion
        cp.spawnSync('sudo', ['-k']);
        process.env.userp = '';
```

from https://github.com/Samsung/ONE-vscode/blob/main/src/Job/JobRunner.ts#L96